cedar-policy-core/src/entities/json/entities.rs

```diff
@@ -153,11 +194,10 @@ impl<'e, 's, S: Schema> EntityJsonParser<'e, 's, S> {
                         }
                     })?,
                 )),
-                Some(schema) => {
-                    // query the schema to get the expected type. Depending on
-                    // the expected type, we may parse the contents of the
-                    // attribute differently.
-                    let (rexpr, expected_ty) = match schema.attr_type(etype, &k) {
+                EntitySchemaInfo::NonAction(desc) => {
+                    // Depending on the expected type, we may parse the contents
+                    // of the attribute differently.
+                    let (rexpr, expected_ty) = match desc.attr_type(&k) {
```

Comment:
[nit] I'd prefer `restricted_expr` here (and below). `rexpr` makes me think `regular_expr`

Reply:
We also have the `val_into_rexpr` and `type_of_rexpr` functions.  This would be a larger refactor if we were to be consistent -- can we do that in a separate PR?